pyg_lib/csrc/sampler/cpu/mapper.h

```diff
@@ -23,26 +25,28 @@ class Mapper {
 
   void fill(const scalar_t* nodes_data, const scalar_t size) {
     if (use_vec) {
-      for (scalar_t i = 0; i < size; ++i)
+      for (scalar_t i = 0; i < size; ++i) {
```

Comment:
Let me post my question here, I read some documents and based on my understanding scalar_t includes both float, double, int32, int64 during compile. But in a lot of our usecases we are iterating over integers. How does pytorch avoid compile float type for these functions? Is there a better way to be more specific to the data types here?

Reply:
There are some helper functions like `is_integral` for a `dtype`, but IMO it is mostly runtime checking. We can also use some STL type checking for compile time.

Reply:
The `AT_DISPATCH_INTEGRAL_TYPES` call handles which types `scalar_t` can take (during compile time).